src/types/UiaTextRangeBase.cpp

```diff
@@ -540,6 +539,14 @@ try
         const auto& buffer = _pData->GetTextBuffer();
         const auto bufferSize = buffer.GetSize();
 
+        // TODO GH#5406: create a different UIA parent object for each TextBuffer
+        // nvaccess/nvda#11428: Ensure our endpoints are in bounds
+        // otherwise, we'll FailFast catastrophically
+        if (!bufferSize.IsInBounds(_start, true) || !bufferSize.IsInBounds(_end, true))
+        {
+            throw E_FAIL;

```

Comment:
This note is me suspiciously checking that all consumers try/catchreturn this private method call.

Reply:
This follow up is me assuring myself that This Is Fine(TM).